src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -78,4 +85,35 @@ public static Map<String, String> parseStringToStringMap(XContentParser parser)
         return map;
     }
 
+    /**
+     * Parse content parser to {@link java.time.Instant}.
+     *
+     * @param parser json based content parser
+     * @return instance of {@link java.time.Instant}
+     * @throws IOException IOException if content can't be parsed correctly
+     */
+    public static Instant parseInstant(XContentParser parser) throws IOException {
+        if (parser.currentToken() == null || parser.currentToken() == XContentParser.Token.VALUE_NULL) {
+            return null;
+        }
+        if (parser.currentToken().isValue()) {
```

Comment:
The method can be simplified like:
```suggestion
        if (parser.currentToken().isValue() && parser.currentToken() != XContentParser.Token.VALUE_NULL) {
             return Instant.ofEpochMilli(parser.longValue());
        }
        
        return null;
```

Reply:
This wouldn't work because we need to have the initial `parser.currentToken() == null` check. if we just executed `parser.currentToken().isValue() ` first then we will get a NullPointerException. I added a UT for this also though, which I didn't have previously

Reply:
You can have the condition like then:
```java
 if (parser.currentToken() != null && parser.currentToken().isValue() && parser.currentToken() != XContentParser.Token.VALUE_NULL)

Reply:
changed